### PR TITLE
検索機能追加実装の叩き

### PR DESCRIPTION
**Pull request by DayBySay**
trello:
https://trello.com/c/kNwf3HHP/698--

### CODE REVIEW

File: PiyoTraffic/guide.html [hunk 1]
```diff
@@ -4,14 +4,42 @@
 <meta charset="UTF-8">
 <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0, user-scalable=0">
 <meta name="format-detection" content="telephone=no" />
-<title>使い方ガイド</title>
+<script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.5.2/jquery.min.js"></script>
```
Comment (co3k) [inline PiyoTraffic/guide.html]: なぜ 2015 年に jQuery 1.5.2 を……

File: PiyoTraffic/guide.html [hunk 2]
```diff
@@ -4,14 +4,42 @@
...
 </head>
 <body>
 <header>
-  <h1>使い方ガイド</h1>
+  <h1>WEB検索</h1>
 </header>
+<div id="searchformBox">
+    <form action="./log.html" id="searchform">
+    <input type="text" name="query" id="keywords" placeholder="映画やドラマ、音楽などを検索">
+        <input type="image" src="./images/search_btn_46x42@2x.png" alt="" id="searchBtn">
```
Comment (tetsuharuohzeki) [inline PiyoTraffic/guide.html]: alt属性がないしインデント崩れてますね

File: PiyoTraffic/guide.html (hunk 1 above)
Comment (co3k) [inline PiyoTraffic/guide.html]: それから jQuery を本当に使わないといけないかどうか、よく考えてみてほしいです。
- この程度の DOM アクセスなら手で書いてもまったく問題なさそう
  - getElementById() とか使いたくなかったとしても IE 8 以上をターゲットにできるなら querySelector 使えばいいし、そうでなくてもクエリセレクタ専用のライブラリを使えばいいかと
- イベントハンドラも IE9 以上をターゲットにできるなら標準の使えばいいし、厳しいならそれ相応のライブラリはあります

File: PiyoTraffic/guide.html [hunk 3]
```diff
@@ -4,14 +4,42 @@
...
+<script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.5.2/jquery.min.js"></script>
+<script type="text/javascript" src="jquery.cookie.js"></script>
+<title>WEB検索</title>
 <link rel="stylesheet" href="reset.css">
 <link rel="stylesheet" href="style.css">
+<script type="text/javascript">
+    $(function(){
+            var cookie_id = "io.github.voyagegroup";
+            if(!$.cookie(cookie_id)) {
+                $("#keywords").attr("placeholder", "通信制限 節約")
```
Comment (co3k) [inline PiyoTraffic/guide.html]: placeholder 属性値は、あくまで記入例だということが伝わるような書き方にしてほしいです

File: PiyoTraffic/guide.html [hunk 4]
```diff
@@ -4,14 +4,42 @@
...
 <meta name="format-detection" content="telephone=no" />
-<title>使い方ガイド</title>
+<script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.5.2/jquery.min.js"></script>
+<script type="text/javascript" src="jquery.cookie.js"></script>
+<title>WEB検索</title>
 <link rel="stylesheet" href="reset.css">
 <link rel="stylesheet" href="style.css">
+<script type="text/javascript">
+    $(function(){
+            var cookie_id = "io.github.voyagegroup";
```
Comment (co3k) [inline PiyoTraffic/guide.html]: cookie の名前は意味のあるものにしてほしいですし、 cookie である時点でターゲットのドメインは自明なのでこの部分の意図が謎な感じです。

File: PiyoTraffic/guide.html [hunk 5]
```diff
@@ -4,14 +4,42 @@
...
+            if(!$.cookie(cookie_id)) {
```
Comment (co3k) [inline PiyoTraffic/guide.html]: たしかに JavaScript から cookie は (僕の知識の範囲では) 扱いにくいですが、この程度のことは普通に文字列マッチングで達成できるような気がします。

File: PiyoTraffic/guide.html [hunk 6]
```diff
@@ -4,14 +4,42 @@
...
+                $("#keywords").attr("placeholder", "通信制限 節約")
+                $.cookie(cookie_id, cookie_id);
+            }
+
+            $("#searchform").bind("submit", function() {
+                if ($("#keywords").val() != "") {
+                    return true;
+                } else {
+                    $("#keywords").attr("placeholder", "キーワードを入力してください。");
```
Comment (co3k) [inline PiyoTraffic/guide.html]: placeholder の使い方としては適切でないと思います。

想定されるアクセス環境も限定されますし、  input 要素に required 属性値を設定すれば充分ではないですかね

File: PiyoTraffic/guide.html (hunk 2 above)
Comment (co3k) [inline PiyoTraffic/guide.html]: このプロジェクトの class 属性値はスネークケースで統一されているようなので、一貫性のために id 属性値もそれにあわせるか、 class 属性値もキャメルケースに変えてください。

File: PiyoTraffic/guide.html (hunk 6 above)
Comment (co3k) [inline PiyoTraffic/guide.html]: ちなみに placeholder 属性の用途については以下のように説明されています。
http://www.w3.org/TR/html5/forms.html#the-placeholder-attribute

> The placeholder attribute represents a short hint (a word or short phrase) intended to aid the user with data entry when the control has no value.

(中略)

> The placeholder attribute should not be used as a replacement for a label. For a longer hint or other advisory text, place the text next to the control.

File: PiyoTraffic/guide.html (hunk 3 above)
Comment (co3k) [inline PiyoTraffic/guide.html]: あ、というかこれは DOM ツリー構築後に実行されないとダメですよね。

DOMContentLoaded が完了したあとに実行するとか、 body 要素の末尾に置くとかを検討してください。

Comment (tetsuharuohzeki) [inline PiyoTraffic/guide.html]: `script[defer]`でも良さそう

File: PiyoTraffic/guide.html (hunk 1 above)
Comment (DayBySay) [inline PiyoTraffic/guide.html]: https://github.com/voyagegroup/tool-apps-terms-privacy/commit/6f30fc032983b2c67550c923a5a75f279a14038c で修正しました。

File: PiyoTraffic/guide.html (hunk 2 above)
Comment (DayBySay) [inline PiyoTraffic/guide.html]: https://github.com/voyagegroup/tool-apps-terms-privacy/commit/8658f02f78435178d8d0a034d0b35da719671ba8 で修正しました。

File: PiyoTraffic/guide.html (hunk 4 above)
Comment (DayBySay) [inline PiyoTraffic/guide.html]: https://github.com/voyagegroup/tool-apps-terms-privacy/commit/6f30fc032983b2c67550c923a5a75f279a14038c で既に訪問済みかどうかを判断している事を示すkey名に修正しました

File: PiyoTraffic/guide.html (hunk 5 above)
Comment (DayBySay) [inline PiyoTraffic/guide.html]: https://github.com/voyagegroup/tool-apps-terms-privacy/commit/6f30fc032983b2c67550c923a5a75f279a14038c でjqueryをやめ、文字列マッチングを用いるよう修正しました。

File: PiyoTraffic/guide.html [hunk 7]
```diff
@@ -4,14 +4,23 @@
...
+</script>
 </head>
 <body>
 <header>
-  <h1>使い方ガイド</h1>
+  <h1>WEB検索</h1>
 </header>
+<div id="search_form_box">
+    <form action="./log.html" id="searchform" onSubmit="return exists_search_keyword()">
+        <input type="text" name="query" id="keywords" placeholder="映画やドラマ、音楽などを検索">
```
Comment (DayBySay) [inline PiyoTraffic/guide.html]: プレースホルダ古いかも

File: PiyoTraffic/style.css [hunk 8]
```diff
@@ -58,16 +68,48 @@ section.step_4{
...
+    position: relative;  
+}  
+#keywords{  
+    width: 260px;  
+    height: 16px;  
+    background-color: #ffffff;  
+    border: 2px solid #765c06;  
+    padding: 14px 14px 14px 14px;
+    color:#662701;
+    -webkit-border-radius: 10px;
```
Comment (tetsuharuohzeki) [inline PiyoTraffic/style.css]: これは`-webkit-` prefixいらない

File: PiyoTraffic/guide.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
+set_placeholeder_if_needed();
```
Comment (tetsuharuohzeki) [inline PiyoTraffic/guide.js]: - 関数の宣言部よりも先にその関数を使うのは、関数の定義位置の巻き上げに依存した挙動なので、よくない
- 先に宣言して、それから使うべき

File: PiyoTraffic/guide.html (hunk 7 above)
Comment (DayBySay) [inline PiyoTraffic/guide.html]: 一旦 `Yahoo！検索` にします

File: PiyoTraffic/guide.js (hunk 9 above)
Comment (DayBySay) [inline PiyoTraffic/guide.js]: @saneyuki 
指摘頂いてから、巻き上げっていう概念は見つけたんですけど[変数の話](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Statements/var)だったのでちゃんと理解できてないかもしれないです。もし公式のドキュメントに記述あれば教えていただけると嬉しいです。

一方で[関数](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Functions_and_function_scope)のサンプルがすべて関数を先に定義してから使ってるので、今後は先に定義するようにします。

File: PiyoTraffic/style.css (hunk 8 above)
Comment (DayBySay) [inline PiyoTraffic/style.css]: 必要なかったので削除しました。 https://github.com/voyagegroup/tool-apps-terms-privacy/commit/6c36f22c326c875ac0efc87c311840613b8ff214

File: PiyoTraffic/style.css [hunk 10]
```diff
@@ -85,6 +127,30 @@ section.step_4{
...
+#search_btn{  
+    position: absolute;  
+    top: 2px;  
+    left: 244px;
+    background: url(images/search_btn_46x42@2x.png) no-repeat;
+    background-size: 46px 42px;
+    width:46px;
+    height:42px;
+    border: none;
+}
```
Comment (co3k) [inline PiyoTraffic/style.css]: 不要な閉じ括弧

File: PiyoTraffic/style.css [hunk 11]
```diff
@@ -3,35 +3,45 @@
...
 	display:block;
 	width:160px;
 	height:31px;
 	background-image : url("./images/title_howto_160x31.png");
 	background-repeat:no-repeat;
-	
-	margin:27px auto -16px; 
+	margin:134px auto -16px; 
+}
+input.red::-webkit-input-placeholder {
```
Comment (co3k) [inline PiyoTraffic/style.css]: このスタイルはもはや使っていないはずなので削除してください

File: PiyoTraffic/style.css [hunk 12]
```diff
@@ -3,35 +3,45 @@
...
+    color:    #f00;
 }
-
 article{
 	display: block;
 	width:100%;
-	padding:18px;
+	padding:18px 18px 18px 18px;
```
Comment (co3k) [inline PiyoTraffic/style.css]: この変更の意図は？

File: PiyoTraffic/style.css (hunk 10 above)
Comment (DayBySay) [inline PiyoTraffic/style.css]: この閉じカッコはメディアクエリの閉じカッコでした。
インデントがついておらず不要な閉じカッコに見えてしまっていました、すいません。
https://github.com/voyagegroup/tool-apps-terms-privacy/commit/31b950f40ec75ac81fdafcccc5fd4b9b5da82375 でインデントをつけて対応しました。

File: PiyoTraffic/style.css (hunk 11 above)
Comment (DayBySay) [inline PiyoTraffic/style.css]: https://github.com/voyagegroup/tool-apps-terms-privacy/commit/16bcecdcf7bf9e23c673afab9222af1c1266a9a2 で削除しました。

File: PiyoTraffic/style.css (hunk 12 above)
Comment (DayBySay) [inline PiyoTraffic/style.css]: https://github.com/voyagegroup/tool-apps-terms-privacy/commit/e36acd3939467a5739f782e4ecb0e4f67cdb6ff4 パディング調整をしてる途中で間違えてコミットしてしまっていたので削除しました。
最終的にパディングの変更はナシです。

**Comment by co3k** [comment]
looks good to me
